core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroup.java

```diff
@@ -112,6 +112,22 @@ protected final void setEndpoints(Iterable<Endpoint> endpoints) {
         completeInitialEndpointsFuture(newEndpoints);
     }
 
+    private static boolean hasChanges(List<Endpoint> oldEndpoints, List<Endpoint> newEndpoints) {
+        if (oldEndpoints.size() != newEndpoints.size()) {
+            return true;
+        }
+
+        for (int i = 0; i < oldEndpoints.size(); i++) {
+            final Endpoint a = oldEndpoints.get(i);
+            final Endpoint b = newEndpoints.get(i);
```

Comment:
Is there any chance that the `newEnpoints.size()` is smaller than `oldEndpoints.size()`?

Reply:
Nope, because we made sure both lists have the same size.